### ISSUE TITLE
Cast to generic type passed to function

### ISSUE THREAD

**Issue by bfelbo** [microsoft/pyright#1158]
**Describe the bug**
We use type checking of content from our users / DB. After type checking, we want the type to be the validated one.

**To Reproduce**
See the code below.

**Expected behavior**
The `validate_type()` function below would return the value as the type passed to it. However, pyright currently reports `Expected class type but received "_T"` error and doesn't acknowledge `confirmed_greeting` as being of type `Greeting`. Is this an issue with pyright or do we need to modify this code?

**Screenshots or Code**
```python
from typing import Literal, Any, TypeVar
import typing

Greeting = Literal["Hi", "Hello"]

# Validate if a variable is in compliance with the type.
# After validation, return it as expected type for static type checking.
_T = TypeVar("_T")
def validate_type(value: Any, expected_type: _T) -> _T:
    # TODO: Add the type checking.. (we use typeguard)
    return typing.cast(expected_type, value)

def send_greeting(msg: Greeting):
    print(msg)

some_str = typing.cast(str, "Hi") # Unknown contents
confirmed_greeting = validate_type(Greeting, some_str) # Should now be Greeting type
send_greeting(confirmed_greeting)
```

**VS Code extension or command-line**
VS Code extension v1.1.84

**Comment by erictraut** [comment]
I think you're on the right track, but you need to make a few minor corrections. First, the type for the parameter `expected_type` should be `Type[_T]`. Second, you need to swap the two arguments to the call to `validate_type`. The second argument should be `Greeting`.
